github/downloader_test.go

```diff
@@ -80,74 +118,380 @@ func getDownloader() (*Downloader, *testutils.Memory, error) {
 	if err != nil {
 		return nil, nil, err
 	}
-
 	storer := new(testutils.Memory)
 	downloader.storer = storer
 	return downloader, storer, nil
 }
 
-func testOnlineRepo(t *testing.T, oracle RepositoryTest, d *Downloader, storer *testutils.Memory) {
+// Tests:
+// 1. Online (live) download with token for git-fixtures org and memory cache
+// 2. Online (live) download with token for git-fixtures repos and memory cache
+// 4. Offline (recorded) download without token for src-d/gitbase repo and memory cache
+// 3. Offline (recorded) download without token for src-d org and memory cache
+// 5. Online (live) download with token for git-fixtures org and DB
+// 6. Online (live) download with token for git-fixtures repos and DB
+// 7. Offline (recorded) download without token for src-d org and DB
+// 8. Offline (recorded) download without token for src-d/gitbase repo and DB
+
+type DownloaderTestSuite struct {
```

Comment:
Maybe I'd have done something like:
```
type DownloaderTestSuite struct {
	suite.Suite
	db         *sql.DB
	downloader *Downloader
	jsonTests  testutils.Tests
}

type DownloaderOnlineMemoryTestSuite struct {
	DownloaderTestSuite
}

type DownloaderOfflineMemoryTestSuite struct {
	DownloaderTestSuite
}

type DownloaderOnlineDBTestSuite struct {
	DownloaderTestSuite
}

type DownloaderOfflineDBTestSuite struct {
	DownloaderTestSuite
}
```

with 2 tests each, one for repo and one for org. Most probably many things for setup could be unified in `DownloaderSuite` instead of repeating them inside each test.

Reply:
Yeah, that is nicer.